src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -139,7 +143,41 @@ else if (authCheckService.isAtLeastTeachingAssistantForExercise(exercise, user))
     }
 
     /**
-     * GET /exercises/:exerciseId : get the "exerciseId" exercise with data useful for tutors.
+     * GET /exercises/:exerciseId/example-solution : get the exercise with example solution without sensitive fields
+     * if the user is allowed to access the example solution and if the example solution is published.
+     *
+     * @param exerciseId the exerciseId of the exercise with the example solution
+     * @return the ResponseEntity with status 200 (OK) and with body the exercise with its example solution after filtering sensitive data, or with
+     * status 404 (Not Found) is the exercise is not found, or with status 403 (Forbidden) if the current user does not have access to the example solution.
+     */
+    @GetMapping("/exercises/{exerciseId}/example-solution")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Exercise> getExerciseForExampleSolution(@PathVariable Long exerciseId) {
+
+        log.debug("REST request to get exercise with example solution: {}", exerciseId);
+
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        Exercise exercise = exerciseRepository.findByIdElseThrow(exerciseId);
+
+        if (exercise.isExamExercise()) {
+            examAccessService.checkExamExerciseForExampleSolutionAccessElseThrow(exercise);
+        }
+        else {
+            // Course exercise
+            if (!authCheckService.isAllowedToSeeExercise(exercise, user)) {
+                throw new AccessForbiddenException("You are not allowed to see this exercise!");
+            }
+            if (!exercise.isExampleSolutionPublished()) {
+                throw new AccessForbiddenException("Example solution for exercise is not published yet!");
+            }
+        }
+
+        exercise.filterSensitiveInformation();
```

Comment:
I'd only filter this if the user is not at least a tutor

Reply:
Since this endpoint is intended to serve the example solution and the instructors can examine the exam summary from the perspective of a student at `/course-management/:courseId/exams/:examId/student-exams/:studentExamId/summary` at the current state I thought it would be better to filter it for instructors and tutors too so that we do not accidentally rely on a data that is not accessible to the students. Also I think if weneed extra data for the instructors or tutors for a future feature it would be easy to just add an extra condition to skip filtering sensitive data according to user role.